lib/src/rules/prefer_asserts_in_initializer_lists.dart

```diff
@@ -31,6 +31,31 @@ class A {
 }
 ```
 
+**GOOD:**
+```dart
+class A {
+  final int a;
+  A(this.a);
+}
+
+class B extends A {
+  B(super.a) : assert(a != 0);
+}
+```
+
+**BAD:**
+```dart
+class A {
+  final int a;
+  A(this.a);
+}
+
+class B extends A {
+  B(super.a) {
+    assert(a != 0);
```

Comment:
I remember seeing some discussion about the scope of `a`, but can't remember now whether it's visible in the body of the constructor. We should check (if you haven't already) because the two examples might not be equivalent (that is, the `a` in the body _might_ be a reference to the getter rather than the parameter).